CenterForOpenScience/ember-osf-web
# [ENG-619] User-search component

- Ticket: https://openscience.atlassian.net/browse/ENG-619
- Feature flag: `N/A`

## Purpose

To add a new user-search view to the contributors component.

## Summary of Changes

- Add new folder structure for `user-search` within the contributors widget
<img width="883" alt="Screen Shot 2021-01-20 at 2 28 29 PM" src="https://user-images.githubusercontent.com/19379783/105233206-86555400-5b37-11eb-9e2b-515529e19dd5.png">

<img width="877" alt="Screen Shot 2021-01-20 at 2 28 43 PM" src="https://user-images.githubusercontent.com/19379783/105233226-8bb29e80-5b37-11eb-9031-7c6d8c13d00b.png">

<img width="881" alt="Screen Shot 2021-01-20 at 2 28 54 PM" src="https://user-images.githubusercontent.com/19379783/105233237-8fdebc00-5b37-11eb-88d6-83b202a34324.png">


## Todo:

- [ ] Unregistered contributor modal
- [ ] Remove selected contributors
- [ ] Save contributors

## Side Effects

`N/A`

## QA Notes

This will only add the user-search part of the add contributor widget. It should be search for users based on `full_name`, `given_name`, `middle_names`, and `family_name`. If there aren't any users, it will show a `No users found` message in the box. If there is an error, it should show `Error, could not load contributors`. And when the list first toggles on, it will show a basic message with instructions on how to use the add contributor widget.
